wp-content/civi-extensions/civirazorpay/cli/update-recurring-contact-info.php

```diff
@@ -0,0 +1,317 @@
+    echo "Status: {$subscription['status']}\n";
+    echo "Subscription ID: {$subscription['id']}\n";
+
+    try {
+      $contactID = $this->handleCustomerDataAndFindContact($subscription);
+
+      if (!$contactID) {
+        throw new Exception("No valid contact could be associated with subscription {$subscription['id']}");
+      }
+
+    }
+    catch (Exception $e) {
+      echo "Error processing subscription {$subscription['id']}: " . $e->getMessage() . "\n";
+    }
+  }
+
+  /**
+   * Handle customer data associated with the subscription.
+   *
+   * @param array $subscription
+   */
+  private function handleCustomerDataAndFindContact(array $subscription) {
+    $subscriptionId = $subscription['id'] ?? NULL;
+    $notes = $subscription['notes'] ?? [];
+    $email = $notes['email'] ?? NULL;
+    $name = $notes['name'] ?? NULL;
+    $mobile = $notes['mobile'] ?? NULL;
+    $address = $notes['address1'] ?? NULL;
+    $panCard = $notes['PAN Card'] ?? NULL;
+
+    try {
+      // Step 1: Fetch first recurring contribution to get contact ID.
+      $firstRecur = ContributionRecur::get(FALSE)
+        ->addSelect('contact_id', 'contribution.id')
+        ->addJoin('Contribution AS contribution', 'LEFT')
+        ->addWhere('processor_id', '=', $subscriptionId)
+        ->addWhere('contribution.source', '=', 'Imported from Razorpay')
+        ->execute()
+        ->first();
+
+      $contactId = $firstRecur['contact_id'] ?? NULL;
+
+      if ($contactId) {
+        echo "Contact found successfully. Contact ID: $contactId\n";
+
+        try {
+          $this->updateDetailsOnContact($contactId, $mobile, $address);
+        }
+        catch (\Exception $e) {
+          \Civi::log()->error('Error updating contact details: ' . $e->getMessage());
+        }
+
+        // Step 2: Now fetch all contributions and update PAN card.
+        try {
+          $this->updateDetailsOnImportedContributions($firstRecur['id'], $panCard);
+        }
+        catch (\Exception $e) {
+          \Civi::log()->error('Error updating PAN card: ' . $e->getMessage());
+        }
+
+        return $contactId;
+      }
+
+    }
+    catch (\Exception $e) {
+      \Civi::log()->error('Error fetching recurring contribution: ' . $e->getMessage());
+    }
+
+    echo "Could not identify a unique contact. Logged for manual intervention.\n";
+    return NULL;
+  }
+
+  /**
+   *
+   */
+  public function updateDetailsOnImportedContributions($id, $panCard) {
+    $allRecurs = Contribution::get(FALSE)
+      ->addSelect('contact_id', 'id')
+      ->addWhere('contribution_recur_id', '=', $id)
+      ->addWhere('is_test', '=', $this->isTest)
+      ->addWhere('source', '=', 'Imported from Razorpay')
+      ->execute();
+
+    foreach ($allRecurs as $record) {
+      $this->updatePanCard([$record], $panCard);
+    }
+  }
+
+  /**
+   *
+   */
+  public function updatePanCard($contributionRecurs, $panCard) {
+    foreach ($contributionRecurs as $item) {
```

Comment:
I don't think you need this `foreach` loop. 

The `foreach ($allRecurs as $record)` in `updateDetailsOnImportedContributions` should be enough.